packages/web-pkg/src/components/CreateShortcutModal.vue

```diff
@@ -6,34 +6,94 @@
       :button-confirm-text="$gettext('Create')"
       :button-confirm-disabled="confirmButtonDisabled"
       @cancel="cancel"
-      @confirm="createShortcut"
+      @confirm="createShortcut(inputUrl, inputFilename)"
+      @keydown.enter="onKeyDownEnter"
     >
       <template #content>
-        url
-        <oc-text-input v-model="inputUrl" />
-        <div class="oc-flex oc-flex-bottom oc-width-1-1 oc-mt-m">
+        <oc-text-input
+          id="create-shortcut-modal-url-input"
+          v-model="inputUrl"
+          :label="$gettext('Shortcut to a webpage or file')"
+        />
+        <oc-drop
+          v-if="showDrop"
+          ref="dropRef"
+          class="oc-pt-s"
+          padding-size="remove"
+          drop-id="create-shortcut-modal-contextmenu"
+          toggle="#create-shortcut-modal-url-input"
+          :close-on-click="true"
+        >
+          <oc-list>
+            <li class="oc-p-xs">
+              <oc-button
+                class="oc-width-1-1"
+                appearance="raw"
+                justify-content="left"
+                @click="dropItemUrlClicked"
+              >
+                <oc-icon name="external-link" />
+                <span v-text="dropItemUrl" />
+              </oc-button>
+            </li>
+            <li v-if="searchTask.isRunning" class="oc-p-xs oc-flex oc-flex-center">
```

Comment:
This and the next `<li>` should be wrapped into a `template` block with a `v-if` that checks if we're inside an authenticated context. Search doesn't work from within public links.